src/apps/chifra/pkg/file/lock.go

```diff
@@ -49,3 +50,28 @@ func changeLock(file *os.File, lockConfig *syscall.Flock_t) error {
 	// More: https://stackoverflow.com/questions/29611352/what-is-the-difference-between-locking-with-fcntl-and-flock
 	return syscall.FcntlFlock(file.Fd(), syscall.F_SETLK, lockConfig)
 }
+
+var DefaultOpenFlags = os.O_RDWR | os.O_CREATE
+var DefaultOpenFlagsAppend = os.O_RDWR | os.O_CREATE | os.O_APPEND
+
+// WaitOnLock tries to lock a file for maximum `maxSecondsLock`
+func WaitOnLock(filePath string, openFlags int) (file *os.File, err error) {
+	file, err = os.OpenFile(filePath, openFlags, 0666)
+	if err != nil {
+		return
+	}
+
+	for i := 0; i < maxSecondsLock; i++ {
+		err = Lock(file)
+		if err == syscall.EAGAIN || err == syscall.EACCES {
+			// Another process has already locked the file, we wait
+			continue
+		}
+		if err != nil {
+			// We cannot lock, so report the error back to the caller
+			return file, err
+		}
+		time.Sleep(1 * time.Second)
+	}
+	return
```

Comment:
If I'm reading this right, we've failed for three seconds to get the lock. Shouldn't this be an error return?